src/elm/query.ts

```diff
@@ -237,8 +237,33 @@ export class Query extends Expression {
   }
 
   exec(ctx: Context) {
+    let sourceResults = this.sources.map(s => s.expression.execute(ctx));
+    // If every source is null, the result is null
+    // See: https://jira.hl7.org/browse/FHIR-40225
+    if (sourceResults.every(r => r == null)) {
+      return null;
+    }
+    const isList = sourceResults.some(r => Array.isArray(r));
```

Comment:
Should we re-use the `typeIsArray` util instead?

Reply:
These days, I think `Array.isArray` is pretty well-supported, so we should probably consider moving all uses of `typeIsArray` to `Array.isArray`.  Starting with this one. ;-)